library/src/scala/quoted/Expr.scala

```diff
@@ -53,8 +143,11 @@ object Exprs {
     override def toString: String = s"Expr(<tasty tree>)"
   }
 
-  /** An Expr representing `'{(~f).apply(~x)}` but it is beta-reduced when the closure is known */
-  final class FunctionAppliedTo[T, +U](val f: Expr[T => U], val x: Expr[T]) extends Expr[U] {
-    override def toString: String = s"Expr($f <applied to> $x)"
+  // TODO Use a List in FunctionAppliedTo(val f: Expr[_], val args: List[Expr[_]])
+  // FIXME: Having the List in the code above trigers an assertion error while testing dotty.tools.dotc.reporting.ErrorMessagesTests.i3187
+  //        This test does redefine `scala.collection`. Further investigation is needed.
+  /** An Expr representing `'{(~f).apply(~x1, ..., ~xn)}` but it is beta-reduced when the closure is known */
+  final class FunctionAppliedTo[+R](val f: Expr[_], val args: Array[Expr[_]]) extends Expr[R] {
```

Comment:
Later, after #5297, we will be able to remove this class altogether as the function application will not be delayed.